### PR TITLE
Define UUID as primary key

### PR DESCRIPTION
**Pull request by JonatascNascimento**
Configure environment to create new tables with `id` column, with `uuid` type to be a primary key.

## Steps
- [x] Create a enable uuid migration
- [x] Set `uuid` as default primary key in rails initializers
- [x] Update all tables, adding `id` column

## Obs

Because of the following error, that got up during migration:
[undefined method `mechanism=' for ActiveSupport::Dependencies:Module (NoMethodError)](https://stackoverflow.com/questions/70849954/undefined-method-mechanism-for-activesupportdependenciesmodule-nomethoder)

I had to remove `gem 'spring-watcher-listen', '~> 2.0.0'` and upgrade `gem spring` to 3.0.0 version.

If is there a problem, please, comment in this pull request.

## Img

Example of a new migration, with id column defined:

image

### CODE REVIEW

**Comment by JonatascNascimento** [comment]
This Pull Request fixes https://github.com/Purple-Stock/open-erp/issues/56 issue

File: db/schema.rb [hunk 1]
```diff
@@ -267,18 +269,6 @@
     t.index ["reset_password_token"], name: "index_users_on_reset_password_token", unique: true
   end
 
-  add_foreign_key "accounts", "users"
   add_foreign_key "active_storage_attachments", "active_storage_blobs", column: "blob_id"
   add_foreign_key "active_storage_variant_records", "active_storage_blobs", column: "blob_id"
```
Comment (JonatascNascimento) [inline db/schema.rb:273]:
```suggestion
  add_foreign_key "active_storage_variant_records", "active_storage_blobs", column: "blob_id"
  add_foreign_key "accounts", "users"
  add_foreign_key "group_products", "groups"
  add_foreign_key "group_products", "products"
  add_foreign_key "products", "categories"
  add_foreign_key "purchase_products", "products"
  add_foreign_key "purchase_products", "purchases"
  add_foreign_key "purchases", "suppliers"
  add_foreign_key "sale_products", "products"
  add_foreign_key "sale_products", "sales"
  add_foreign_key "sales", "customers"
  add_foreign_key "simplo_items", "products"
  add_foreign_key "simplo_items", "simplo_orders"
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -31,10 +31,12 @@ jobs:
           RAILS_ENV: test
         run: |
           sudo apt-get -yqq install libpq-dev
+          yarn install --check-files
           gem install bundler -v 2.2.29
           bundle install
-          bundle exec rails db:create
-          bundle exec rails db:migrate
-          bundle exec rails db:seed
+          bin/rails db:environment:set RAILS_ENV=development
```
Comment (JonatascNascimento) [inline .github/workflows/main.yml:37]:
```suggestion
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -31,10 +31,12 @@ jobs:
...
+          bin/rails db:create
```
Comment (JonatascNascimento) [inline .github/workflows/main.yml:38]:
```suggestion
          bin/rails db:create
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -31,10 +31,11 @@ jobs:
           RAILS_ENV: test
         run: |
           sudo apt-get -yqq install libpq-dev
+          yarn install --check-files
           gem install bundler -v 2.2.29
```
Comment (JonatascNascimento) [inline .github/workflows/main.yml:35]:
```suggestion
          yarn install --check-files
          bin/rails db:environment:set RAILS_ENV=development
          gem install bundler -v 2.2.29
```

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -31,10 +31,12 @@ jobs:
           RAILS_ENV: test
         run: |
           sudo apt-get -yqq install libpq-dev
+          yarn install --check-files
+          bin/rails db:environment:set RAILS_ENV=development
```
Comment (JonatascNascimento) [inline .github/workflows/main.yml:35]:
```suggestion

```

**Review by puppe1990 — APPROVED** [review]

**Pull request merged**